web/src/app/shared/components/tree/tree.component.ts

```diff
@@ -61,36 +68,56 @@ export class TaskanaTreeComponent implements OnInit, AfterViewChecked, OnDestroy
 
   private filterTextOld: string;
   private filterIconOld = '';
-  private removedNodeIdSubscription: Subscription;
+  private action: ACTION;
+  private destroy$ = new Subject<void>();
 
   constructor(
-    private treeService: TreeService,
     private elementRef: ElementRef,
     private classificationsService: ClassificationsService,
+    private location: Location,
+    private store: Store,
+    private notificationsService: NotificationService,
   ) {
   }
 
   @HostListener('document:click', ['$event'])
   onDocumentClick(event) {
     if (this.checkValidElements(event) && this.tree.treeModel.getActiveNode()) {
-      this.unSelectActiveNode();
+      this.deselectActiveNode();
     }
   }
 
   ngOnInit() {
-    this.removedNodeIdSubscription = this.treeService.getRemovedNodeId().subscribe(value => {
-      const removedNode = this.getNode(value);
-      if (removedNode.parent) {
-        removedNode.parent.collapse();
+    this.activeAction$.pipe(takeUntil(this.destroy$)).subscribe(action => {
+      this.action = action;
+    });
+
+    this.selectedClassificationId$.pipe(takeUntil(this.destroy$)).subscribe(selectedClassificationId => {
+      this.selectNodeId = typeof selectedClassificationId !== 'undefined' ? selectedClassificationId : undefined;
```

Comment:
Should we filter undefined values out so that we don't have to do this check anymore?